### PR TITLE
Added starring for repositories

### PR DESCRIPTION
**Pull request by ssergienko**
Added starring for repositories. It works well for me. Have a look.

### CODE REVIEW

**Comment by aendrew** [comment]
Hi! Thanks for the PR! Please add tests and I'll merge. Thanks!

**Comment by ssergienko** [comment]
Hi, guys. Added the tests.

**Comment by aendrew** [comment]
@ssergienko Mind looking at the six failures? Also, please squash your commits. Thanks!

**Comment by ssergienko** [comment]
Hello, mate. Am sorry, it didn't look for me like that. Think I didn't rose environment correctly. 
The result for "npm install" looks like this - http://prntscr.com/7w2drs 
Could you give me an advice, how to run this tests or where could I read about?

**Comment by aendrew** [comment]
@ssergienko My, that's irritating. The test runner I use, `argg`, expects `cat` to exist, which it does not in Windows.

Try removing line 12 from package.json and running `npm install` again. Then run `tape tests/*.js` to run tests (`npm test` won't work because it uses `argg`).

**Comment by aendrew** [comment]
Ref #205.

@ssergienko Want to take another stab at this? I'm going to remove the dependency on argg in 0.10.7, would be great to get this into 0.10.8. :smile:

**Comment by aendrew** [comment]
@ssergienko If you rebase against master, you should be able to run your tests locally. Try this:

```
git remote add upstream git@github.com:michael/github.git
git fetch upstream
git rebase -i upstream/master
```

You should then be able to do `npm test` without problem.

**Comment by aendrew** [comment]
Closed in error; reopening.

**Comment by AurelioDeRosa** [comment]
Hi @ssergienko. Are you still interested in updating this API?

File: test/test.repo.js [hunk 1]
```diff
@@ -170,3 +170,43 @@ test('Repo Returns commit errors correctly', function(t) {
     t.end();
   });
 });
+
+test('Repo starring works correctly', function(t) {
+  var timeout = setTimeout(function () {
+      t.fail();
```
Comment (AurelioDeRosa) [inline test/test.repo.js]: We aren't using tap anymore so the tests should be updated.

File: github.js [hunk 2]
```diff
@@ -724,6 +724,24 @@
           }
           _request("GET", url, null, cb);
       };
+
+      // starring
+      this.isStarred = function(owner, repo, cb) {
+        _request("GET", "/user/starred/" + owner + '/' + repo, null, function(err,res) {
```
Comment (AurelioDeRosa) [inline github.js]: You can have `cb` as the last parameter so you can avoid creating an anonymous function that just invoke the callback.

Comment (ssergienko) [inline github.js]: Hi there

No I have no time to work with it. Please use it in any way you like.

Best regards
Serge

Comment (AurelioDeRosa) [inline github.js]: Thank you for the quick reply.

**Comment by AurelioDeRosa** [comment]
Closing this PR as these methods have been introduced with commits 63b62ca0c6b02fd1ba62d0f33740641a9dd3531a, 1fd209d55efad56ad1812642a39a4e3a2ff50813, and 15b69aa54f8ee628d7d08f0fe94854cbd9351fad.

**Pull request closed without merging**